modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -71,7 +72,9 @@ def _read(cls, filepath_or_buffer, **kwargs):
         elif index_col is None and not usecols:
             # When names is set to some list that is smaller than the number of columns
             # in the file, the first columns are built as a hierarchical index.
-            empty_pd_df = pandas.read_csv(filepath_or_buffer, nrows=0)
+            empty_pd_df = pandas.read_csv(
```

Comment:
Why changed only `elif index_col is None and not usecols:` branch?

Reply:
In `if names is None` branch, the parameters for the call are created from `kwargs`, that include `encoding` parameter.